isaric/parsers/isaric-core.toml

```diff
@@ -0,0 +1,1708 @@
+    { field = "heparint_cmyn", description = "Heparin?", ref = "Y/N/NK", "can_skip" = true },
+    { field = "ccm_b_heparint_cmyn", description = "Heparin", ref = "Y/N/NK" },
+  ]
+
+[visit.treatment_inhaled_nitric_oxide]
+  combinedType = "any"
+  fields = [
+    { field = "inhalednit_cmtrt", description = "Inhaled Nitric Oxide?", ref = "Y/N/NK" },
+    { field = "daily_nitritc_cmtrt", description = "Inhaled Nitric Oxide ?", ref = "Y/N/NK" },
+  ]
+
+[visit.treatment_inotropes_vasopressors]
+  combinedType = "any"
+  description = "A vasopressor is a pharmaceutical agent that cause vasoconstriction."
+  fields = [
+    { field = "inotrop_cmtrt", ref = "Y/N/NK" },
+    { field = "daily_inotrope_cmyn", ref = "Y/N/NK" },
+  ]
+
+[visit.treatment_neuromuscular_blocking_agents]
+  field = "daily_neuro_cmtrt"
+  description = "Neuromuscular blocking agents?"
+  ref = "Y/N/NK"
+
+[visit.treatment_other]
+  combinedType = "set"
+  excludeWhen = "none"
+  fields = [
+    { field = "othertx_cmtrt_1", description = "<i>Other Agent 1</i> - Agent:" },
+    { field = "othertx_cmtrt_3", description = "<i>Other Agent 3</i> - Agent:", "can_skip" = true },
+    { field = "othertx_cmtrt_2", description = "<i>Other Agent 2</i> - Agent:" },
+    { field = "daily_other_prtrt" },
+    { field = "other_cm", "can_skip" = true },
+  ]
+
+[visit.treatment_prone_position]
+  combinedType = "any"
+
+  [[visit.treatment_prone_position.fields]]
+    field = "pronevent_prtrt"
+    description = "Prone ventilation refers to ventilation with the patient lying in the prone position. If the patient received prone ventilation at any time during their hospital stay, please tick YES and indicate the total duration in days."
+    ref = "Y/N/NK"
+
+  [[visit.treatment_prone_position.fields]]
+    field = "daily_prone_cmtrt"
+    description = "Prone ventilation refers to ventilation with the patient lying in the prone position. If the patient received prone ventilation at any time during their hospital stay, please tick YES and indicate the total duration in days."
+    ref = "Y/N/NK"
+
+  [[visit.treatment_prone_position.fields]]
+    field = "ccm_b_pronevent_prtrt"
+    description = "Prone Positioning?"
+    ref = "Y/N/NK"
+
+[visit.date_outcome]
+  field = "dsstdtc"
+  description = "Please state the date for the outcome listed above."
+
+[visit.outcome]
+  field = "dsterm"
+  description = "Please state the date for the outcome listed above."
+  values = { 1 = "discharged", 2 = "hospitalised", 3 = "transferred", 4 = "death", 5 = "palliative discharge" }
+
+  ## OBSERVATION
+
+[[observation]]
+  name = "abdominal_pain"
+  phase = "admission"
+  date = { ref = "admissionDateHierarchy" }
+  is_present = { field = "abdopain_ceoccur_v2", values = { 1 = true, 2 = false }, can_skip = true }
+
+[[observation]]
+  name = "abdominal_pain"
+  phase = "followup"
+  date = { field = "flw_survey_date" }
+  is_present = { field = "flw_stomach_pain", values = { 1 = true, 0 = false } }
```

Comment:
@pipliggins a note more in reference to the main CORE data, symptoms reported at follow up refer to any symptom experienced in the last 7 days, not necessarily at the time of the follow up survey completion. I remember a while back discussing using duration_days to indicate a timing interval such as this, should this be added? The exception is for flw_fever where the timing of the last fever is a mapped response from flw_last_fever. This is also applicable to the RAPID parser follow up fields, and the ISARIC follow ups (not yet reviewed)

Reply:
i see these edits for params = [7] - this is reflecting a date in the 7 days prior to flw_survey_date, is that right?

Reply:
yep!